fastai/vision/learner.py

```diff
@@ -95,119 +96,114 @@ def unet_learner(data:DataBunch, arch:Callable, pretrained:bool=True, blur_final
-        truthlabels=np.asarray(self.y_true, dtype=int) 
-        classes_ids=[k for k in enumerate(self.data.classes)]
-        predclass=np.asarray(self.pred_class)
-        for i, pred in enumerate(predclass):
-            where_truth=np.nonzero((truthlabels[i]>0))[0]
-            mismatch=np.all(pred!=where_truth)
-            if mismatch: 
-                mismatches_idxs.append(i)
-                if l_dim > 1 : losses_mismatches.append((losses[i][pred],i))
-                else: losses_mismatches.append((losses[i],i))
-            if l_dim > 1: infotup=(i, pred, where_truth, losses[i][pred], np.round(self.probs[i], decimals=3)[pred], mismatch)
-            else: infotup=(i, pred, where_truth, losses[i], np.round(self.probs[i], decimals=3)[pred], mismatch)
-            infolist.append(infotup)
-        ds = self.data.dl(self.ds_type).dataset
-        mismatches = ds[mismatches_idxs]
-        ordlosses=sorted(losses_mismatches, key = lambda x: x[0], reverse=True)
-        for w in ordlosses: ordlosses_idxs.append(w[1])
-        mismatches_ordered_byloss=ds[ordlosses_idxs]
-        print(str(len(mismatches))+' misclassified samples over '+str(len(self.data.valid_ds))+' samples in the validation set.')
-        for ima in range(len(mismatches_ordered_byloss)):
-            mismatchescontainer.append(mismatches_ordered_byloss[ima][0]) 
-        for sampleN in range(samples):
-            actualclasses=''
-            for clas in infolist[ordlosses_idxs[sampleN]][2]:
-                actualclasses=actualclasses+' -- '+str(classes_ids[clas][1])
-            imag=mismatches_ordered_byloss[sampleN][0]
-            imag=show_image(imag, figsize=figsz)
-            imag.set_title(f"""Predicted: {classes_ids[infolist[ordlosses_idxs[sampleN]][1]][1]} \nActual: {actualclasses}\nLoss: {infolist[ordlosses_idxs[sampleN]][3]}\nProbability: {infolist[ordlosses_idxs[sampleN]][4]}""",
-                           loc='left')
-            plt.show()
-            if save_misclassified: return mismatchescontainer
-
-    def confusion_matrix(self, slice_size:int=None):
-        "Confusion matrix as an `np.ndarray`."
-        x=torch.arange(0,self.data.c)
-        if slice_size is None: cm = ((self.pred_class==x[:,None]) & (self.y_true==x[:,None,None])).sum(2)
-        else:
-            cm = torch.zeros(self.data.c, self.data.c, dtype=x.dtype)
-            for i in range(0, self.y_true.shape[0], slice_size):
-                cm_slice = ((self.pred_class[i:i+slice_size]==x[:,None])
-                            & (self.y_true[i:i+slice_size]==x[:,None,None])).sum(2)
-                torch.add(cm, cm_slice, out=cm)
-        return to_np(cm)
-
-    def plot_confusion_matrix(self, normalize:bool=False, title:str='Confusion matrix', cmap:Any="Blues", norm_dec:int=2,
+def top_losses(self, k:int=None, largest=True):
```

Comment:
This could go in the general `ClassificationInterpretation` too, it's not specific to vision.